core/src/main/scala/zio/config/ReadError.scala

```diff
@@ -138,7 +138,7 @@ sealed trait ReadError[A] extends Exception { self =>
 
         case _ => format(sequence).updated(0, "╥")
       }
-    }).mkString("\n")
+    }).mkString(System.lineSeparator())
```

Comment:
Should we be using `zio.lineSeparator` ?

Advantage: Seamless integration with zio
Disadvantage: Currently zoo-config allows users to retrieve config as a simple Either[ReadError, A]. It won't be user-friendly if they end up having to deal with effect system just to prettyPrint their errors.

@jdegoes  @senia-psm  Any views on this?

Reply:
Probably ZIO should change to make line separator static, assuming it is guaranteed to be static on the JVM. In the meantime, fine to use System.lineSeparator() here.